app/backend/src/couchers/jobs/worker.py

```diff
@@ -0,0 +1,61 @@
+"""
+Background job workers
+"""
+
+import logging
+import traceback
+from time import sleep
+
+from couchers.db import session_scope
+from couchers.jobs import JOBS
+from couchers.models import BackgroundJob, BackgroundJobState, BackgroundJobType
+from couchers.utils import now
+
+logger = logging.getLogger(__name__)
+
+# Note that the database stuff here isn't perfect, it has some issues with having multiple parallel servicers
+# See: https://www.2ndquadrant.com/en/blog/what-is-select-skip-locked-for-in-postgresql-9-5/
+
+
+def process_job(job_id):
+    with session_scope() as session:
+        # grab the job
+        job = (
+            session.query(BackgroundJob).filter(BackgroundJob.id == job_id).filter(BackgroundJob.is_ready).one_or_none()
+        )
+        # if it's gone, too bad
+        if not job:
+            return
+
+        # mark it as ours
+        job.state = BackgroundJobState.working
+        job.try_count += 1
+        session.commit()
```

Comment:
Is it possible that at the same moment another `process_job` is executing these same lines for the same job because two `service_jobs` functions selected the same job?

Reply:
Yes, it's broken at the moment